src/main/java/fi/hsl/transitdata/metroats/MetroEstimatesFactory.java

```diff
@@ -0,0 +1,228 @@
+            metroKey = TransitdataProperties.formatMetroId(stopNumber, startDatetime);
+        } else {
+            log.warn("Failed to get stop number for stop shortNames: startStopShortName: {}, endStopShortName: {}", startStopShortName, endStopShortName);
+            return Optional.empty();
+        }
+        MetroAtsProtos.MetroEstimate.Builder metroEstimateBuilder = MetroAtsProtos.MetroEstimate.newBuilder();
+
+        // Set fields from mqtt-pulsar-gateway into metroEstimateBuilder
+        metroEstimateBuilder.setSchemaVersion(metroEstimateBuilder.getSchemaVersion());
+        // trainType
+        Optional<MetroAtsProtos.MetroTrainType> maybeMetroTrainTypeAts = getMetroTrainTypeAts(metroEstimate.trainType);
+        if (!maybeMetroTrainTypeAts.isPresent()) {
+            log.warn("metroEstimate.trainType is missing: {}", metroEstimate.trainType);
+            return Optional.empty();
+        }
+        metroEstimateBuilder.setTrainType(maybeMetroTrainTypeAts.get());
+        // journeySectionprogress
+        Optional<MetroAtsProtos.MetroProgress> maybeMetroAtsProgress = getMetroAtsProgress(metroEstimate.journeySectionprogress);
+        if (!maybeMetroAtsProgress.isPresent()) {
+            log.warn("metroEstimate.journeySectionprogress is missing: {}", metroEstimate.journeySectionprogress);
+            return Optional.empty();
+        }
+        metroEstimateBuilder.setJourneySectionprogress(maybeMetroAtsProgress.get());
+        metroEstimateBuilder.setBeginTime(metroEstimate.beginTime);
+        metroEstimateBuilder.setEndTime(metroEstimate.endTime);
+        metroEstimateBuilder.setStartStopShortName(startStopShortName);
+
+        // Set fields from Redis into metroEstimateBuilder
+        Optional<Map<String, String>> metroJourneyData = getMetroJourneyData(metroKey);
+        if(!metroJourneyData.isPresent()) {
+            log.warn("Couldn't read metroJourneyData from redis. Metro key: {}, redis map: {}.", metroKey, metroJourneyData);
+            return Optional.empty();
+        }
+        metroJourneyData.ifPresent(map -> {
+            if (map.containsKey(TransitdataProperties.KEY_OPERATING_DAY))
+                metroEstimateBuilder.setOperatingDay(map.get(TransitdataProperties.KEY_OPERATING_DAY));
+            if (map.containsKey(TransitdataProperties.KEY_START_STOP_NUMBER))
+                metroEstimateBuilder.setStartStopNumber(map.get(TransitdataProperties.KEY_START_STOP_NUMBER));
+            if (map.containsKey(TransitdataProperties.KEY_START_TIME))
+                metroEstimateBuilder.setStartTime(map.get(TransitdataProperties.KEY_START_TIME));
+            if (map.containsKey(TransitdataProperties.KEY_DVJ_ID))
+                metroEstimateBuilder.setDvjId(map.get(TransitdataProperties.KEY_DVJ_ID));
+            if (map.containsKey(TransitdataProperties.KEY_START_STOP_SHORT_NAME))
+                metroEstimateBuilder.setStartStopShortName(map.get(TransitdataProperties.KEY_START_STOP_SHORT_NAME));
```

Comment:
This is not in Redis so it can be removed.

Reply:
Ah yes, forgot to remove it from here as I removed it from redis